File: vigenere.c
#import <stdio.h>
#import <cs50.h>
#import <string.h>
#import <stdlib.h>
#import <ctype.h>

int main(int argc, string argv[])
{
    string keyword = argv[1];
    if (argc != 2)
    {
        printf("Give program a single keyword for cipher!\n");
        return 1;
    }
    else
    {
        int k = 0;
        int keys[strlen(keyword)];

        for (int i = 0; i < strlen(keyword); i++)
        {
            if (isalpha(keyword[i]) == 0)
            {
                printf("Give program a single keyword for cipher!\n");
                return 1;
            }
            else
            {
                keyword[i] = tolower(keyword[i]);
                keys[i] = ((int)keyword[i] - 97) % 26;
            }
        }
        string text = get_string("plaintext: ");

        for (int i = 0; i <= strlen(text); i++)
        {
            if (k == strlen(keyword))
            {
                k = 0;
            }
            if (isalpha(text[i]))
            {
                if (isupper(text[i]))
                {
                    text[i] = (((int)text[i] - 65 + keys[k]) % 26) + 65;
                    k++;
                }
                if (islower(text[i]))
                {
                    text[i] = (((int)text[i] - 97 + keys[k]) % 26) + 97;
                    k++;
                }
            }
        }
        printf("ciphertext: %s\n", text);
        return 0;
    }
}

File: pset6/vigenere/vigenere.py
import sys

if (len(sys.argv) != 2) or sys.argv[1].isalpha() == False:
    print("Give program a single keyword for cipher!")
    exit(1)
else:
    keyword = sys.argv[1]
    text = input("plaintext:")
    print("ciphertext:", end="")
    k = 0
    for i in text:
        if (k == len(keyword)):
            k = 0
        if i.isalpha():
            # get shift to convert from acsii to alphabetical order
            if i.islower():
                shift = 97
            elif i.isupper():
                shift = 65
        # character((ascii value - shift + cypher in lowercase) modulo 26 to get alpabetical order value) + shift to get back to ascii)
            i = chr(((ord(i) - shift + (ord(keyword[k].lower()) - 97) % 26) % 26) + shift)
            k += 1
        print(i, end='')
print()
